airflow/gcp/hooks/gcs.py

```diff
@@ -323,6 +323,26 @@ def delete(self, bucket_name, object_name):
 
         self.log.info('Blob %s deleted.', object_name)
 
+    def delete_bucket(self, bucket_name: str):
+        """
+        Delete an bucket object from the Google Cloud Storage
+
+        :param bucket_name: name of the bucket which will be deleted
+        :type bucket_name: str
+        """
+
+        client = storage.Client()
+        bucket = client.bucket(bucket_name)
+
+        if not bucket.exists():
```

Comment:
It is better to assume the optimistic variant, i.e. delete bucket, but if it is not exists to catch an exception. In this way, we have fewer requests.

Reply:
But int this case we catch 404 and should we assume it means "no bucket"?

Reply:
like this?
`        try:
            bucket.delete()
        except NotFound as err:
            self.log.error(err)`
404 will be thrown if there is no bucket:
`ERROR - 404 DELETE https://storage.googleapis.com/storage/v1/b/test-gcs-example-bucket: Not Found`